packages/server/src/api/routes/tests/oauth2.spec.ts

```diff
@@ -92,4 +97,94 @@ describe("/oauth2", () => {
       ])
     })
   })
+
+  describe("update", () => {
+    let existingConfigs: OAuth2Config[] = []
+
+    beforeEach(async () => {
+      existingConfigs = []
+      for (let i = 0; i < 10; i++) {
+        const oauth2Config = makeOAuth2Config()
+        const result = await config.api.oauth2.create(oauth2Config)
+
+        existingConfigs.push({ ...oauth2Config, id: result.config.id })
+      }
+    })
+
+    it("can update an existing configuration", async () => {
+      const { id: configId, ...configData } = _.sample(existingConfigs)!
+
+      await config.api.oauth2.update(configId, {
+        ...configData,
+        name: "updated name",
+      })
+
+      const response = await config.api.oauth2.fetch()
+      expect(response.configs).toHaveLength(existingConfigs.length)
+      expect(response.configs).toEqual(
+        expect.arrayContaining([
+          {
+            id: configId,
+            name: "updated name",
+            url: configData.url,
+          },
+        ])
+      )
+    })
+
+    it("throw if config not found", async () => {
+      await config.api.oauth2.update("unexisting", makeOAuth2Config(), {
+        status: 404,
+        body: { message: "OAuth2 config with id 'unexisting' not found." },
+      })
+    })
+
+    it("throws if trying to use an existing name", async () => {
+      const [config1, config2] = _.sampleSize(2, existingConfigs)
+      const { id: configId, ...configData } = config1
+
+      await config.api.oauth2.update(
+        configId,
+        {
+          ...configData,
+          name: config2.name,
+        },
+        {
+          status: 400,
+          body: { message: "Name is not available." },
+        }
+      )
+    })
+  })
+
+  describe("delete", () => {
+    let existingConfigs: OAuth2Config[] = []
+
+    beforeEach(async () => {
+      existingConfigs = []
+      for (let i = 0; i < 5; i++) {
+        const oauth2Config = makeOAuth2Config()
+        const result = await config.api.oauth2.create(oauth2Config)
+
+        existingConfigs.push({ ...oauth2Config, id: result.config.id })
+      }
+    })
+
+    it("can delete an existing configuration", async () => {
+      const { id: configId } = _.sample(existingConfigs)!
+
+      await config.api.oauth2.delete(configId, { status: 204 })
+
+      const response = await config.api.oauth2.fetch()
```

Comment:
Isn't this potentially flaky?

If the test below runs before this one, 5 configs get created and 0 deleted. Then this test runs, `existingConfigs` is reset to `[]` and 5 more configs get created. There are now 10 configs for the tenant. Then you delete 1 and compare the total to 4, but there are 9.

What am I missing?

Reply:
It should not. First, jest test run always in order per each file. And we are creating new tenants per each run, so each test is independent